src/api/Base.js

```diff
@@ -165,54 +193,152 @@ class Base {
         }
     };
 
-    /**
-     * Generic GET request
-     *
-     * @param {string} id the file id
-     * @param {Object} params the query params
-     */
-    getData(id: string, params?: Object): Promise<Object> {
-        return this.xhr.get({
-            id: getTypedFileId(id),
-            url: this.getUrl(id),
-            params
-        });
-    }
-
     /**
      * Gets the URL for the API, meant to be overridden
-     * @param {string} id the file id
+     * @param {string} id - The file id
      */
     /* eslint-disable no-unused-vars */
     getUrl(id: string) {
         /* eslint-enable no-unused-vars */
         throw new Error('Implement me!');
     }
 
+    /**
+     * Attaches success and error callbacks for a given API call
+     * @param {Function} successCallback - Success function
+     * @param {Function} errorCallback - Error function
+     */
+    attachCallbacks(successCallback: Function, errorCallback: Function) {
+        if (typeof successCallback === 'function') {
```

Comment:
Due to the fact that flow checks that the arguments are a function, this helper function isnt necessary